Dockerfile

```
@@ -65,4 +67,45 @@ RUN mkdir /root/sgx-rust && \
 # TODO: remove once all the code supports the later nightly toolchains
 RUN /root/.cargo/bin/rustup toolchain install nightly-2021-03-25
 
+FROM teaclave-base AS builder
+ENV SGX_SDK=/opt/sgxsdk
 WORKDIR /root
+
+COPY --from=teaclave-base /root/sgx-rust/ ./sgx-rust/
+COPY --from=teaclave-base /root/.cargo/ ./.cargo/
+
+WORKDIR /root/rtc-data
+ARG SGX_MODE=HW
+
+COPY . .
```

It might be nice to insert a separate `cargo fetch` step here, for the bandwidth-constrained among us? 😄️

```suggestion
COPY . .

RUN --mount=type=cache,target=/root/.cargo/registry \
    --mount=type=cache,target=/root/.cargo/git \
    . /root/.cargo/env && cargo fetch
    
```